### DISCUSSION TITLE
Strict equality for literals

### DISCUSSION THREAD

**Discussion by vadimcn** [microsoft/pyright#1671] [Q&A]
Is it possible to have Pyright detect bogus comparisons with literals?    For example:
```python
OS = Literal['Linux', 'Darwin', 'Windows']
os: OS = 'Linux'

# Sometime later
if os == 'linux':
    ...
```

I don't want to declare `OS` as enum, because an existing API uses strings.

**Comment by erictraut** [comment]
This is a good suggestion. I've created a feature request using your example: https://github.com/microsoft/pyright/issues/1674
